bot/exts/moderation/infraction/management.py

```diff
@@ -201,29 +201,26 @@ async def infraction_edit(
     # region: Search infractions
 
     @infraction_group.group(name="search", aliases=('s',), invoke_without_command=True)
-    async def infraction_search_group(self, ctx: Context, query: t.Union[UserMention, Snowflake, str]) -> None:
+    async def infraction_search_group(self, ctx: Context, query: t.Union[MemberOrUser, Snowflake, str]) -> None:
         """Searches for infractions in the database."""
         if isinstance(query, int):
             await self.search_user(ctx, discord.Object(query))
-        else:
+        elif isinstance(query, str):
             await self.search_reason(ctx, query)
+        else:
+            await self.search_user(ctx, query)
 
     @infraction_search_group.command(name="user", aliases=("member", "id"))
-    async def search_user(self, ctx: Context, user: t.Union[discord.User, proxy_user]) -> None:
+    async def search_user(self, ctx: Context, user: t.Union[MemberOrUser, discord.Object]) -> None:
         """Search for infractions by member."""
         infraction_list = await self.bot.api_client.get(
             'bot/infractions/expanded',
             params={'user__id': str(user.id)}
         )
 
-        user = self.bot.get_user(user.id)
-        if not user and infraction_list:
-            # Use the user data retrieved from the DB for the username.
-            user = infraction_list[0]["user"]
-            user = escape_markdown(user["name"]) + f"#{user['discriminator']:04}"
```

Comment:
Is it not better to display this information over just an ID?

Reply:
This does still give this information just via an isinstance check instead (see the new [L221](https://github.com/python-discord/bot/blob/remove-redundant-converters/bot/exts/moderation/infraction/management.py#L221)). 

I am about to do a commit to add the `escape_markdown` though as I forgot that, which edits how this works to
```py
        if isinstance(user, (discord.Member, discord.User)):
            user_str = escape_markdown(user.name) + user.discriminator
        else:
            user_str = str(user.id)
```

Reply:
@TizzySaurus You misunderstood me. The lines I commented were only used when a `discord.Object` was passed. In that case, all it has is an ID, and it sucks to only display an ID. Instead, it relies on the last known information in the database, if it exists. If not, then it can fall back to just an ID.

Furthermore, what you had previously was sufficient when it's a `Member` or `User`. They have already have a `__str__` which does exactly what you changed it to.

Reply:
Right yes, I understand your point now of displaying just id vs name#discrim. I'll add this functionality back.

In regards to the `Member`/`User`'s `__str__` method, it doesn't escape any markdown which is why I put it to this (what it was before I changed anything).

Reply:
Wouldn't `escape_markdown(str(user))` be simpler at least?

Reply:
I suppose we'd want something like
```py
        if isinstance(user, (discord.Member, discord.User)):
            user_str = escape_markdown(user.name) + user.discriminator
        else:  # is discord.Object instance
            if infraction_list:  # can get the name#discrim from DB
                user = infraction_list[0]["user"]
                user_str = escape_markdown(user["name"]) + f"#{user['discriminator']:04}"
            else:  # can't get name from DB so just use the id
                user_str = str(user.id)
```? (without the comments)

Reply:
Yes but I believe the first case can be simplified to `escape_markdown(str(user))` (you forgot the `#` anyway).

Reply:
Right yes, `escape_markdown(str(user))` would be easier so I'll go with that.
```py
        if isinstance(user, (discord.Member, discord.User)):
            user_str = escape_markdown(str(user))
        else:  # is discord.Object instance
            if infraction_list:  # can get the name#discrim from DB
                user = infraction_list[0]["user"]
                user_str = escape_markdown(user["name"]) + f"#{user['discriminator']:04}"
            else:  # can't get name from DB so just use the id
                user_str = str(user.id)
```like this?

Reply:
Yes, that _looks_ right. Test it for good measure before checking it in.

Reply:
Pushed :+1: